wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceRuntimes.java

```diff
@@ -293,27 +292,25 @@ public void stop(String workspaceId, Map<String, String> options) throws NotFoun
         }
         eventService.publish(DtoFactory.newDto(WorkspaceStatusEvent.class)
                                        .withWorkspaceId(workspaceId)
-                                       .withPrevStatus(WorkspaceStatus.RUNNING)
-                                       .withStatus(WorkspaceStatus.STOPPING));
+                                       .withPrevStatus(state.status)
+                                       .withStatus(STOPPING));
 
         try {
             state.runtime.stop(options);
 
             // remove before firing an event to have consistency between state and the event
             runtimes.remove(workspaceId);
-            eventService.publish(DtoFactory.newDto(WorkspaceStatusEvent.class)
-                                           .withWorkspaceId(workspaceId)
-                                           .withPrevStatus(WorkspaceStatus.STOPPING)
-                                           .withStatus(STOPPED));
+            publishWorkspaceStoppedEvent(workspaceId, STOPPING, null);
         } catch (InfrastructureException e) {
             // remove before firing an event to have consistency between state and the event
             runtimes.remove(workspaceId);
-            eventService.publish(DtoFactory.newDto(WorkspaceStatusEvent.class)
-                                           .withWorkspaceId(workspaceId)
-                                           .withPrevStatus(WorkspaceStatus.STOPPING)
-                                           .withStatus(STOPPED)
-                                           .withError("Error occurs on workspace runtime stop. Error: " +
-                                                      e.getMessage()));
+            if (!(e instanceof RuntimeStartInterruptedException)) {
+                publishWorkspaceStoppedEvent(workspaceId,
+                                             STOPPING,
+                                             "Error occurs on workspace runtime stop. Error: " + e.getMessage());
+            } else {
+                publishWorkspaceStoppedEvent(workspaceId, STOPPING, null);
```

Comment:
Can you elaborate in which cases stop can throw RuntimeStartInterruptedException?